Makefile

```diff
@@ -1,125 +1,280 @@
+
+# Commands
+PIPENV_RUN := pipenv run
+PYTHON := $(PIPENV_RUN) python3
+PIP := $(PIPENV_RUN) pip3
+YAPF := $(PIPENV_RUN) yapf
+ISORT := $(PIPENV_RUN) isort
+PYLINT := $(PIPENV_RUN) pylint
+SHELLCHECK := $(PIPENV_RUN) shellcheck
+
+# check if environment file exists
+ifneq ("$(wildcard $(ENV_FILE))","")
+include ${ENV_FILE}
+else
+$(error "environment file does not exist ${ENV_FILE}")
+endif
+
+# check db access
+DB_ACCESS := $(shell pg_isready -h ${PGHOST} &> /dev/null && echo true || echo false)
+ifeq ($(DB_ACCESS),false)
+$(warning ${RED}we need a valid postgres connection for this correct use of makefile! connection to '$(PGHOST)' was not successful${RESET})
+endif
+
+# Maintenance / Index Commands
+# EFS Index will be mounted as bind mount
+export DOCKER_EXEC :=  docker run \
+				--rm \
+				-t \
+				-v $(SPHINX_EFS):/var/lib/sphinxsearch/data/index/ \
+				--env-file $(ENV_FILE) \
+				--name $(DOCKER_LOCAL_TAG)_maintenance \
+				$(DOCKER_IMG_LOCAL_TAG)
+
+export DOCKER_EXEC_LOCAL :=  docker run \
+				--rm \
+				-t \
+				-v $(CURRENT_DIR)/conf/:/var/lib/sphinxsearch/data/index/ \
+				--env-file $(ENV_FILE) \
+				--name $(DOCKER_LOCAL_TAG)_maintenance \
+				$(DOCKER_IMG_LOCAL_TAG)
+
+# AWS variables
+AWS_DEFAULT_REGION = eu-central-1
+
+INDEX ?= 'Set INDEX variable to specify the index or index prefix to create'
+DB ?= 'Set DB variable to specify the database pattern for the index creation'
 
 .PHONY: help
 help:
 	@echo "Usage: make <target>"
 	@echo
 	@echo "Possible targets:"
 	@echo
-	@echo "Indexing only for updates (sudo su sphinxsearch):"
+	@echo "${BOLD}${BLUE}Setup TARGETS${RESET}"
+	@echo "- setup                     Create the python virtual environment and activate it"
+	@echo "- dev                       Create the python virtual environment with developper tools and activate it"
+	@echo "- ci                        Create the python virtual environment and install requirements based on the Pipfile.lock"
 	@echo
-	@echo "- index-all                 Update all indices (does NOT re-create config file)"
-	@echo "- index-grep                Update indices that match a given pattern. Pass the pattern as IPATTERN=mypattern directly on the commandline"
-	@echo "- index-search              Update swisssearch indices (does NOT re-create config file)"
-	@echo "- index-layer               Update all the layers indices (does NOT re-create config file)"
-	@echo "- index-feature             Update all the features indices (does NOT re-create config file)"
-	@echo "- move-template             Move template to the apropriate locations"
+	@echo "${BOLD}${BLUE}Formating, linting targets${RESET}"
+	@echo "- format                    Format the python source code"
+	@echo "- ci-check-format           Format the python source code and check if any files has changed. This is meant to be used by the CI."
+	@echo "- lint                      Lint the python source code"
+	@echo "- shellcheck                shellcheck all the bash scripts"
```

Comment:
Maybe it would be better to have the two in one target checking both ? So when working you don't have to bother running both when you change both python and bash code, simply run one.